stablehlo/dialect/Base.cpp

```diff
@@ -261,67 +261,96 @@ FailureOr<std::pair<int64_t, int64_t>> inferMergedDimAndBound(
   return std::make_pair(size, bound);
 }
 
-// TODO(zhouxin) Refactor to better handle errors and return single type
-LogicalResult inferMostSpecificType(
+// Inference rules for conditional branches (lhs/rhs are commutative):
+//       Dim of lhs     Dim of rhs      Infer
+//  c0:  X              X               X
+//  c1:  X              ?               ?
+//  c2:  X              ?, B(>=X)       ?, B(>=X)
+//  c3:  X              ?, B(<X)        ?, B(<X)
+//  c4:  ?              ?               ?
+//  c5:  ?              ?, B            ?, B
```

Comment:
should C5 be,

? <=B to ?

If we end up taking the first branch, then the size could be larger.

Reply:
Agreed.